tests/tt_eager/python_api_testing/unit_testing/test_complex.py

```diff
@@ -799,3 +799,30 @@ def test_level2_is_imag(bs, memcfg, dtype, device, function_level_defaults):
         passing, output = comp_pcc(tt_cpu, tt_dev)
     logger.info(output)
     assert passing
+
+@pytest.mark.parametrize(
+    "memcfg",
+    (
+        ttl.tensor.MemoryConfig(ttl.tensor.TensorMemoryLayout.INTERLEAVED, ttl.tensor.BufferType.DRAM),
+        ttl.tensor.MemoryConfig(ttl.tensor.TensorMemoryLayout.INTERLEAVED, ttl.tensor.BufferType.L1),
+    ),
+    ids=["out_DRAM", "out_L1"],
+)
+@pytest.mark.parametrize("dtype", ((ttl.tensor.DataType.BFLOAT16,)))
+@pytest.mark.parametrize("bs", ((1, 1), (1, 2), (2, 2)))
+def test_level2_polar(bs, memcfg, dtype, device, function_level_defaults):
+    input_shape = torch.Size([bs[0], bs[1], 32, 64])
+    # check real
+    x = Complex(input_shape)
+    x = Complex(None,x.real,x.imag)
+    
+    xtt = ttl.tensor.complex_tensor(
+        ttl.tensor.Tensor(x.abs, dtype).to(ttl.tensor.Layout.ROW_MAJOR).to(device, memcfg),
+        ttl.tensor.Tensor(x.angle, dtype).to(ttl.tensor.Layout.ROW_MAJOR).to(device, memcfg),
+    )
```

Comment:
My comment was unrelated to the current implementation issues or accuracy.

It was just based on what seems like unnecessarily creating the input tensors for both pt and tt in a complex wrapper when they are/should just be regular tensors. For example the line `tt_cpu = torch.polar(x.real, x.imag)` the inputs should be the abs and angle, and returns a complex tensor with real and imag values, and I find it confusing that you are basically passing a complex tensor as input when it's not supposed to be a real complex tensor.

This is overall minor since it is not related to implementation so not blocking the pr on this.